flux/src/main/java/com/groupbyinc/quickstart/controller/NavigationController.java

```diff
@@ -45,46 +44,45 @@
     private HashMap<String, Query> queryQueue = new HashMap<String, Query>();
     private HashMap<String, CloudBridge> bridgeQueue = new HashMap<String, CloudBridge>();
 
-    @RequestMapping(value = "/json.html")
-    public @ResponseBody
-    String getMoreNavigations(@RequestParam String navigationName, HttpServletRequest request)
-            throws IOException, JspException {
+    @RequestMapping(value = "/moreRefinements.html")
+    ModelAndView getMoreNavigations(@RequestParam String navigationName, @RequestParam String selectedRefinements,
+                                    HttpServletRequest request) throws IOException, JspException {
 
         String clientKey = getCookie(request, "clientKey", "").trim();
 
-        StringBuilder sb = new StringBuilder();
-
         Query query = queryQueue.get(clientKey);
         CloudBridge bridge = bridgeQueue.get(clientKey);
 
-        if(query == null || bridge == null) {
-            return sb.toString();
-        }
-
         navigationName = navigationName.trim();
-        List<Navigation> navigations = null;
 
-        if(MapUtils.isNotEmpty(query.getNavigations())) {
-            navigations = new ArrayList<Navigation>(query.getNavigations().values());
+        Map<String, Object> model = new HashMap<String, Object>();
+        Navigation availableNavigation = new Navigation().setName(navigationName);
+
+        if(query == null || bridge == null) {
+            model.put("results", null);
+            model.put("nav", availableNavigation);
+            return new ModelAndView("includes/navLink.jsp", model);
         }
 
+        Results results = new Results();
+        results.setSelectedNavigation(Utils.getSelectedNavigations(selectedRefinements));
         RefinementsResult refinementsResults = bridge.refinements(
-                new Query().setArea(query.getArea())
-                           .addRefinementsByString(query.getRefinementString()) //
-                           .setCollection(query.getCollection()) //
-                           .addValueRefinement("gbi_stream_upload", "1"), navigationName); //
```

Comment:
@effervescentia So I removed it and ran over the tests for edge cases Will had me do. The error doesn't show up anymore. It might have been a combination of removing this and another bug that caused the error so once I resolved the other bugs this one resolved as well